kairon/shared/data/collection_processor.py

```
@@ -286,5 +287,66 @@ def delete_collection(bot: str, name: str):
+                    "index": index,
+                    "error": str(e)
+                })
+
+        if errors:
+            raise AppException(f"Errors in bulk insert: {errors}")
+
+        if collection_docs:
+            try:
+                CollectionData.objects.insert(collection_docs)
+            except Exception as e:
+                raise AppException(f"Bulk insert failed: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)